pkg/controllers/nodetemplate/suite_test.go

```diff
@@ -362,4 +367,171 @@ var _ = Describe("AWSNodeTemplateController", func() {
+							{Key: aws.String("Name"), Value: aws.String("test-ami-3")},
+							{Key: aws.String("foo"), Value: aws.String("bar")},
+						},
+					},
+				},
+			})
+		})
+		It("should resolve amiSelector AMIs and requirements into status", func() {
+			nodeTemplate.Spec.AMISelector = nil
+			ExpectApplied(ctx, env.Client, nodeTemplate)
+			ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
+			nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
+			amiIDsInStatus := lo.Map(nodeTemplate.Status.AMIs, func(ami v1alpha1.AMI, _ int) string {
+				return ami.Name
+			})
+			sort.Strings(amiIDsInStatus)
+			expectedNameOutput := []string{"amazon-linux-2", "amazon-linux-2-arm64", "amazon-linux-2-gpu"}
+			Expect(amiIDsInStatus).To(Equal(expectedNameOutput))
+			expectedRequirementsOutput := [][]v1.NodeSelectorRequirement{
+				{
+					{
+						Key:      v1.LabelArchStable,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{v1alpha5.ArchitectureAmd64},
+					},
+					{
+						Key:      v1alpha1.LabelInstanceAcceleratorManufacturer,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{string(v1alpha1.AWSAcceleratorManufacturer), string(v1alpha1.NVIDIAacceleratorManufacturer)},
+					},
+				},
+				{
+					{
+						Key:      v1.LabelArchStable,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{v1alpha5.ArchitectureArm64},
+					},
+					{
+						Key:      v1alpha1.LabelInstanceAcceleratorManufacturer,
+						Operator: v1.NodeSelectorOpNotIn,
+						Values:   []string{string(v1alpha1.AWSAcceleratorManufacturer), string(v1alpha1.NVIDIAacceleratorManufacturer)},
+					},
+				},
+				{
+					{
+						Key:      v1.LabelArchStable,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{v1alpha5.ArchitectureAmd64},
+					},
+					{
+						Key:      v1alpha1.LabelInstanceAcceleratorManufacturer,
+						Operator: v1.NodeSelectorOpNotIn,
+						Values:   []string{string(v1alpha1.AWSAcceleratorManufacturer), string(v1alpha1.NVIDIAacceleratorManufacturer)},
+					},
+				},
+			}
+			amiRequirementsInStatus := lo.Map(nodeTemplate.Status.AMIs, func(ami v1alpha1.AMI, _ int) []v1.NodeSelectorRequirement {
+				return ami.Requirements
+			})
+			for ind := range amiRequirementsInStatus {
+				sort.Slice(amiRequirementsInStatus[ind], func(i, j int) bool {
+					return amiRequirementsInStatus[ind][i].Key > amiRequirementsInStatus[ind][j].Key
+				})
+			}
+			Expect(amiRequirementsInStatus).To(Equal(expectedRequirementsOutput))
+
+		})
+		It("Should resolve a valid AMI selector", func() {
+			ExpectApplied(ctx, env.Client, nodeTemplate)
+			ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
+			nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
+			amiIDsInStatus := lo.Map(nodeTemplate.Status.AMIs, func(ami v1alpha1.AMI, _ int) string {
+				return ami.Name
+			})
+			sort.Strings(amiIDsInStatus)
+			Expect(nodeTemplate.Status.AMIs).To(Equal(
```

Comment:
optional: Use `ContainsElements` for consistency and to ensure that order doesn't matter